Drop testnet3

Sjors opened the pull request:
Testnet3 was deprecated in v28 with the introduction of testnet4. It will be dropped in v30 _or later_, please discuss in #31975.

There is no more `[test]` network section. The user is expected to set testnet4 explicitly, which should make future rotation of testnets easier.

If a `[test]` or `[testnet3]` section is present in `bitcoin.conf` a warning is issued:

```
Warning: .../bitcoin.conf:18 Section [test] is not recognized.
```

Starting with `-testnet3` will be greeted with an error message:

```
Error: Testnet3 was dropped in version 30.0. Consider using '-testnet4' instead.
```

This also prevents a fallback to mainnet.

Starting with `-test` was already not possible with testnet3.

Review hints:
- make a (Windows) Guix build and test the (un)installer

Guix hashes:

```
(todo)
```

instagibbs commented:
I've said it elsewhere, but I think it deserves a quite long deprecation cycle, along with some data gathering.

We don't want services not upgrading for fixes because their test environment would break.

e.g., what exchanges with testnet environments have adopted testnet4 (or signet), much less would be ok with removing testnet3?

Does btcd support it? LND? Eclaire? CLN? etc.

Lots of work to be done here imo.

Sjors commented:
Let's open a separate RFC to discuss when to drop testnet3. I can rebase this for a while if we decide v30 is too early.

Sjors committed:
Drop testnet3 seed nodes

Sjors committed:
gui: drop testnet3

Sjors committed:
Drop testnet3 from docs and dev tools

Also use "test network" instead of "testnet" where appropriate (e.g.
because it also applies to signet).

Sjors committed:
validation: drop testnet3 comment about BIP34

fjahr commented:
Concept ACK

Sjors commented:
I split 5d6aa13cf047e31005afca919ca9cf181e30905b into smaller commits. If we decide to punt this until after v30 then I'll PR some of these separately. E.g. there's some tests that can use testnet4 instead of testnet3 and perhaps we can drop GUI support earlier.

Sjors committed:
Disallow -testnet3

Abort node startup and inform user they should use -testnet4.
Modify the deprecation message test to cover this.

Remove testnet3 references from bitcoin-cli documentation
and bash helper.

Sjors committed:
consensus: drop testnet3

This drops the chain parameters and ChainType::TESTNET.

Sjors commented:
Will rebase after #31649 lands.

Sjors commented:
Rebased after #31649.

Sjors commented on test/functional/wallet_crosschain.py:
```diff
@@ -20,10 +20,7 @@ def skip_test_if_missing_module(self):
-        self.nodes[1].chain = 'testnet3'
-        self.nodes[1].extra_args = ['-maxconnections=0', '-prune=550'] # disable testnet sync
-        self.nodes[1].replace_in_config([('regtest=', 'testnet='), ('[regtest]', '[test]')])
+        # Node 1 is omitted, it was previously used with testnet3.
```
I made a note to split this off, unless someone else gets to it first.

Sjors replied:
#32088

yancyribbens commented on contrib/zmq/zmq_sub.py:
```diff
@@ -34,7 +34,7 @@
     sys.exit(1)
 
-port = 28332
+port = 48332
```
The [wiki](https://github.com/bitcoin/bips/blob/master/bip-0094.mediawiki) says the default port is `48333`.  Is this a typo?

Sjors replied:
48333 is the default p2p port

48332 is used for RPC, which is not defined in the BIP because it's specific to Bitcoin Core (as is ZMQ).

Sjors committed:
test: drop testnet3 coverage

Except for the deprecation warning in feature_config_args.py.
This test now writes its own config file, which was previously done by
test_acceptstalefeeestimates_arg_support.

The deprecation warning test will be removed in a later commit.

Sjors commented:
Rebased after #32088 and #32057, which both simplify this PR.

hodlinator commented on doc/bitcoin-conf.md:
```diff
@@ -41,7 +41,7 @@ Comments may appear in two ways:
 
 Network specific options can be:
-- placed into sections with headers `[main]` (not `[mainnet]`), `[test]` (not `[testnet]`, for testnet3), `[testnet4]`, `[signet]` or `[regtest]`;
+- placed into sections with headers `[main]` (not `[mainnet]`), `[testnet4]`, `[signet]` or `[regtest]`;
```
Could this be part of #32096?

Another part is probably better to change at least as early:
https://github.com/bitcoin/bitcoin/blob/b3162d10ea93cdec8e04150289ff61d6fed9a8e9/doc/bitcoin-conf.md?plain=1#L27

Probably better to use "regtest=1" or "signet=1".

Sjors replied:
This lists all the options available to the user, so I think it belongs here.

I'll look into your suggestion on the next rebase.

Sjors replied:
Changed to `signet=1`, guess that could have gone into #32096.

hodlinator commented on contrib/linearize/example-linearize.cfg:
```diff
@@ -7,9 +7,6 @@ host=127.0.0.1
 #mainnet default
 port=8332
 
-#testnet default
-#port=18332
-
```
Noticed while looking at #32096 that this file was untouched there.

Testnet3 has ***rpc***port=18332 according to how chainparamsbase.cpp sets `CBaseChainParams(..., rpc_port)`...
This config is setting the P2P ports for all networks so that they will collide with the default RPC ports and prevent successful startup (verified with the mainnet config line). Might as well just remove all ports IMO.

Sjors replied:
I don't really understand what this file is doing. Can you open a fresh PR to fix it?

hodlinator replied:
I was mistaken, this config file is meant to be consumed by the linearize-(hashes|data).py scripts, not bitcoind. Using `port` in this context is referring to which port the RPC interface is expected to be running on when one connects as a client. Please resolve this thread.

hodlinator commented on src/kernel/chainparams.cpp:
```diff
@@ -191,101 +191,6 @@ class CMainParams : public CChainParams {
-        consensus.vDeployments[Consensus::DEPLOYMENT_TESTDUMMY].nStartTime = Consensus::BIP9Deployment::NEVER_ACTIVE;
-        consensus.vDeployments[Consensus::DEPLOYMENT_TESTDUMMY].nTimeout = Consensus::BIP9Deployment::NO_TIMEOUT;
-        consensus.vDeployments[Consensus::DEPLOYMENT_TESTDUMMY].min_activation_height = 0; // No activation delay
-
```
(Inline thread in random position)

nit: src/external_signer.h references "test" and never "testnet4" which seems to be the argument the code would be sending in.

Sjors replied:
I assume you're talking about the `chain` argument? This is passed directly to HWI (or equivalent software). I'll update the comment on the next rebase to swap "test" for "testnet4" which was missed when testnet4 was launched.

hodlinator replied:
Exactly, from what I could see it would be set from `ArgsManager::GetChainTypeString()` in one case and `CChainParams::GetChainTypeString()` in another.

Sjors commented on test/functional/feature_assumeutxo.py:
```diff
@@ -103,7 +103,6 @@ def expected_error(msg):
         invalid_magics = [
             # magic, name, real
             [0xf9beb4d9, "main", True],
-            [0x0b110907, "test", True],
```
Indeed this should move.

(it's handy if you mention the commit when making a comment)

hodlinator commented on src/kernel/chainparams.cpp:
```diff
@@ -191,101 +191,6 @@ class CMainParams : public CChainParams {
     }
 };
 
```
(Inline comment at random position):

Noticed doc/REST-interface.md still mentions testnet3 port:
https://github.com/bitcoin/bitcoin/blob/4e6e73a3b2c5c3a4a07683bc3a6b59f8e27b78f3/doc/REST-interface.md?plain=1#L6-L7

Sjors replied:
Fixed

Sjors committed:
doc: use "test nework" in external_signer.h

Sjors committed:
doc: use signet=1 in bitcoin-conf.md example

Sjors commented:
Rebased after #32096.

Sjors commented:
Rebased after #29039.

@instagibbs how long do you think we should wait for this? https://github.com/bitcoin/bitcoin/pull/31974#issuecomment-2694932831

My [mailinglist post](https://gnusha.org/pi/bitcoindev/4B21A2CA-2224-4BD1-AA5D-C5B2F8859E4B@sprovoost.nl/T/#t) didn't result in very passionate pleas for keeping it. And although some projects do still use it, they haven't demonstrated a serious intention to upgrade anytime soon.

Should we perhaps wait for testnet5 to see if that fixes the issues we see on testnet4? I suppose we have enough ports...

Or should I mark this ready for review and just drop testnet3 in v30?

Sjors commented:
Trivial rebase after #24450.

Sjors commented:
Rebased after #33274.

Sjors commented:
Trivial rebase after #32579 (touching `chainparams.cpp`).

Sjors committed:
kernel: drop testnet3

Sjors commented:
Rebased after #30595 and #33702.

Sjors committed:
doc: add release note

fanquake commented:
I'm going to close this for now. It's been open for > a year, and is basically just rebasing mechanical change. There doesn't seem to be any consensus on dropping Testnet 3, Testnet 4 is still broken, and various proposals to fix it don't seem to be getting traction (yet?), and there's discussion of a Testnet 5. I think discussion about when to drop Testnet 3 can continue in #31975 (and this PR is linked if someone wants to look at the changes), and these changes can be revived if that happens.
